src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUninstantiatedInternalClassesTests.cs

```diff
@@ -1471,6 +1471,44 @@ internal class MyTextBoxDesigner { }
 }");
         }
 
+        [Fact, WorkItem(3199, "https://github.com/dotnet/roslyn-analyzers/issues/3199")]
+        public async Task CA1812_AliasingTypeNewConstraint_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+using System;
+
+namespace SomeNamespace
+{
+    public class MyAliasType<T>
+        where T : class, new()
+    {
+        public static void DoSomething() {}
+    }
+
+    internal class C {}
+}",
+                        @"
+using MyAliasOfC = SomeNamespace.MyAliasType<SomeNamespace.C>;
```

Comment:
Consider adding a test which the original fix wouldn't have caught, i.e. `A<B<C>>` should also mark `C` as used.

Reply:
Sure!